src/_pytest/pytester.py

```diff
@@ -676,12 +692,11 @@ def to_text(s):
             p.write_text(source.strip(), encoding=encoding)
             if ret is None:
                 ret = p
-        if ret is not None:
-            ret = py.path.local(ret)
+        assert ret is not None
```

Comment:
I wasn't brave enough to add this assert, but it eliminates all of the return types of the `make*` methods so :+1: from me.